### PR TITLE
Migration to tf-nix, in-housed kat's lib functions

### PR DESCRIPTION
**Pull request by kittywitch**

### CODE REVIEW

File: ci/niv-cron.nix [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                if [[ -n $OPENSSH_PRIVATE_KEY ]]; then
+                  git add nix/sources.json
+                  export GIT_{COMMITTER,AUTHOR}_EMAIL=github@kittywit.ch
+                  export GIT_{COMMITTER,AUTHOR}_NAME="niv cron job"
+                  git commit --message="ci: niv update"
+                  if [[ $GITHUB_REF = refs/heads/main ]]; then
+                    GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" \
+                      git push ssh://gitea@git.kittywit.ch:62954/kat/nixfiles.git HEAD:main
+                  fi
+                fi
```
Comment (u1f408) [inline ci/niv-cron.nix:107]: - make this use this github repo as authoritative 
- change committer email
- change cachix thingies

File: ci/niv-cron.nix [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+              echo "no source changes" >&2
+            fi
+          '';
+        impure = true;
+      };
+  };
+
+  ci.gh-actions.checkoutOptions.submodules = false;
+
+  cache.cachix = {
```
Comment (u1f408) [inline ci/niv-cron.nix:121]: - add my own cachix things here

File: ci/nodes.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      targets = main.deploy.targets;
+      enabledTargets = filterAttrs (_: v: v.enable) main.deploy.targets;
+      enabledHosts = concatLists (mapAttrsToList (targetName: target: target.nodeNames) enabledTargets);
+    in
+    mapAttrs' (k: nameValuePair "${k}") (genAttrs enabledHosts (host: {
+      tasks.${host}.inputs = channels.nixfiles.network.nodes.${host}.deploy.system;
+    }));
+
+  ci.gh-actions.checkoutOptions.submodules = false;
+  cache.cachix.arc = {
```
Comment (u1f408) [inline ci/nodes.nix:31]: - add my own cachix things here

File: config/modules/meta/imports.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    network.importing = {
+      nixosImports = mkDefault (map (path: toString path) [
+        (root + "/config/hosts/HN/nixos.nix")
+        (root + "/config/trusted/hosts/HN/nixos.nix")
+      ]);
+      homeImports = mkDefault (map (path: toString path) [
+        (root + "/config/hosts/HN/home.nix")
+        (root + "/config/trusted/hosts/HN/home.nix")
+      ]);
+      users = mkDefault (singleton "kat");
```
Comment (u1f408) [inline config/modules/meta/imports.nix]: - `s/kat/iris`

File: config/targets/common/default.nix [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{ config, ... }: {
+  commandPrefix = "pass";
+  folderPrefix = "secrets";
+  folderDivider = "/";
+}
```
Comment (u1f408) [inline config/targets/common/default.nix:5]: @kittywitch how does this work and can i make it use `bitw`

**Review by u1f408 — CHANGES_REQUESTED** [review]
like, y'know, nya

(thank you)

File: config/targets/common/default.nix (hunk 5 above)
Comment (kittywitch) [reply config/targets/common/default.nix:5]: commandPrefix "bitw get", in bitw the folder you keep a secret in is folderPrefix, keep folder divider the same

**Pull request closed without merging**